src/Persistence/MassTransit.Azure.Cosmos/AzureCosmos/NewtonsoftJsonCosmosClientFactory.cs

```diff
@@ -19,14 +20,16 @@ public class NewtonsoftJsonCosmosClientFactory :
         IDisposable
     {
         readonly CosmosAuthSettings _authSettings;
+        readonly CosmosClientOptions _clientOptions;
         readonly ConcurrentDictionary<Type, Lazy<CosmosClient>> _clients;
 
-        public NewtonsoftJsonCosmosClientFactory(CosmosAuthSettings authSettings)
+        public NewtonsoftJsonCosmosClientFactory(CosmosAuthSettings authSettings, IOptions<CosmosClientOptions> clientOptions)
         {
             if (authSettings == null)
                 throw new ArgumentNullException(nameof(authSettings));
 
             _authSettings = authSettings;
+            _clientOptions = clientOptions.Value;
```

Comment:
GetService can return `null`, yet the options value is not checked. Also, I don't see the options added anywhere in the configuration.

Reply:
This was checked with @NooNameR and he asked me to remove the null check as MT calls AddOptions 
https://github.com/MassTransit/MassTransit/blob/3db6cafad9a536a45648eeb79bbd8d6a2e424ff0/src/MassTransit/Configuration/DependencyInjection/DependencyInjectionRegistrationExtensions.cs#L169-L169
So `GetService` cannot return null. I removed the check. Do you want me to put it back?

Reply:
I missed, just use `GetRequiredService` to make @phatboyg happy

Reply:
Well, if you're going to make it required, you should add it.

Reply:
It doesn't call `AddOptions<CosmosClientOptions>()` though.

Reply:
I replaced `GetService` by `GetRequiredService`. For the explicit `services.Configure<CosmosClientOptions>` requested by @phatboyg, I let you discuss together and will be able to implement what you finally want.

Reply:
I'll just [drop this here](https://learn.microsoft.com/en-us/dotnet/core/extensions/options-library-authors) to understand the guidance from Microsoft.

Reply:
you can look here:
https://github.com/dotnet/runtime/blob/bee65d373e98b6cec39abec54d2eade98b0a03c6/src/libraries/Microsoft.Extensions.Options/src/OptionsServiceCollectionExtensions.cs#L226-L242
if you look into next releases, they do not even have generic `AddOptions<T>()`

Reply:
also, we have Instrumentation without Options, which works:
https://github.com/MassTransit/MassTransit/blob/6f0a00a984d5eb6db9a49256dbe4736fdb9ca460/src/MassTransit/Configuration/DependencyInjection/DependencyInjectionRegistrationExtensions.cs#L164

but if you wanna be safe, I do not mind tho

Reply:
@Baltima can you add `.AddOptions< CosmosClientOptions>` call please?

Reply:
Of course. I'll try to find the best place to add the call and ping you when it is ready to be reviewed.

Reply:
I'd recommend to add it here, most likely:
https://github.com/MassTransit/MassTransit/blob/3fc05c29400f1000f3ae73cfb9dfdd0d8e343e86/src/Persistence/MassTransit.Azure.Cosmos/Configuration/Configuration/CosmosSagaRepositoryConfigurator.cs#L160

Reply:
Done